code/controllers/subsystem/minimaps.dm

```diff
@@ -268,3 +270,117 @@ SUBSYSTEM_DEF(minimaps)
 	images_by_source -= source
 	removal_cbs[source].Invoke()
 	removal_cbs -= source
+
+
+/**
+ * Fetches a /obj/screen/minimap instance or creates on if none exists
+ * Note this does not destroy them when the map is unused, might be a potential thing to do?
+ * Arguments:
+ * * zlevel: zlevel to fetch map for
+ * * flags: map flags to fetch from
+ */
+/datum/controller/subsystem/minimaps/proc/fetch_minimap_object(zlevel, flags)
+	var/hash = "[zlevel]-[flags]"
+	if(hashed_minimaps[hash])
+		return hashed_minimaps[hash]
+	var/obj/screen/minimap/map = new(null, zlevel, flags)
+	hashed_minimaps[hash] = map
+	return map
+
+///Default HUD screen minimap object
+/obj/screen/minimap
+	name = "Minimap"
+	icon = null
+	icon_state = ""
+	layer = ABOVE_HUD_LAYER
+	screen_loc = "1,1"
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/obj/screen/minimap/Initialize(mapload, target, flags)
+	. = ..()
+	icon = SSminimaps.minimaps_by_z["[target]"].hud_image
+	SSminimaps.add_to_updaters(src, flags, target)
+
+
+/**
+ * Action that gives the owner access to the minimap pool
+ */
+/datum/action/minimap
+	name = "Toggle Minimap"
+	action_icon_state = "minimap"
+	///Flags to allow the owner to see others of this type
+	var/minimap_flags = MINIMAP_FLAG_ALL
+	///marker flags this will give the target, mostly used for marine minimaps
+	var/marker_flags = MINIMAP_FLAG_ALL
+	///boolean as to whether the minimap is currently shown
+	var/minimap_displayed = FALSE
+	///Minimap object we'll be displaying
+	var/obj/screen/minimap/map
+
+/datum/action/minimap/Destroy()
+	map = null
+	return ..()
+
+/datum/action/minimap/action_activate()
+	. = ..()
+	if(!map)
+		return
+	if(minimap_displayed)
+		owner.client.screen -= map
+	else
+		owner.client.screen += map
+	minimap_displayed = !minimap_displayed
```

Comment:
probably could do that with early return

Reply:
An early return doesnt apply here